src/orion/core/io/experiment_builder.py

```diff
@@ -0,0 +1,279 @@
+import logging
+import os
+
+import orion
+from orion.core.io import resolve_config
+from orion.core.io.database import Database, DuplicateKeyError
+from orion.core.worker.experiment import Experiment, ExperimentView
+
+
+log = logging.getLogger(__name__)
+
+
+class ExperimentBuilder(object):
+    """Builder for :class:`orion.core.worker.experiment.Experiment`
+    and :class:`orion.core.worker.experiment.ExperimentView`
+
+    .. seealso::
+
+        `orion.core.io.experiment_builder` for more information on the process of building
+        experiments.
+
+        :class:`orion.core.worker.experiment.Experiment`
+        :class:`orion.core.worker.experiment.ExperimentView`
+    """
+
+    # pylint:disable=no-self-use
+    def fetch_default_options(self):
+        """Get dictionary of default options"""
+        return resolve_config.fetch_default_options()
+
+    # pylint:disable=no-self-use
+    def fetch_env_vars(self):
+        """Get dictionary of environment variables specific to Oríon"""
+        return resolve_config.fetch_env_vars()
+
+    def fetch_file_config(self, cmdargs):
+        """Get dictionary of options from configuration file provided in command-line"""
+        return resolve_config.fetch_config(cmdargs)
+
+    def fetch_local_config(self, cmdargs):
+        """Get dictionary of options from all local sources (not from db)
+
+        .. seealso::
+
+            `orion.core.io.experiment_builder` for more information on the hierarchy of
+            configurations.
+        """
+        default_options = self.fetch_default_options()
+        env_vars = self.fetch_env_vars()
+        cmdconfig = self.fetch_file_config(cmdargs)
+
+        return resolve_config.merge_configs(default_options, env_vars, cmdconfig, cmdargs)
+
+    def fetch_db_config(self, cmdargs):
+        """Get dictionary of options from all local sources (not from db)
+
+        Note
+        ----
+            This method builds an experiment view in the background to fetch the configuration from
+            the database.
+        """
+        try:
+            experiment_view = self.build_view_from(cmdargs)
+        except ValueError as e:
+            if "No experiment with given name" in str(e):
+                return {}
+
+        return experiment_view.configuration
+
+    def fetch_full_config(self, cmdargs):
+        """Get dictionary of the full configuration of the experiment.
+
+        .. seealso::
+
+            `orion.core.io.experiment_builder` for more information on the hierarchy of
+            configurations.
+
+        Note
+        ----
+            This method builds an experiment view in the background to fetch the configuration from
+            the database.
+        """
+        default_options = self.fetch_default_options()
+        env_vars = self.fetch_env_vars()
+        db_config = self.fetch_db_config(cmdargs)
+        cmdconfig = self.fetch_file_config(cmdargs)
+
+        exp_config = resolve_config.merge_configs(
```

Comment:
I really don't like duplicating all these calls to the fetching function. Couldn't we modify `fetch_local_config` with a boolean defaulted to False that indicates if we want to merge the database with the other ones? I feel like this would be cleaner.

Reply:
I wanted to make the methods as obvious as possible since the process is confusing. Maybe you are right and I went a bit too far, too verbose. I would agree to remove `fetch_local_config(self, cmdargs)` and turn `fetch_full_config(self, cmdargs)` into `fetch_config(self, cmdargs, use_db=True)`. I would make `True` by default the optional attribute `use_db` since you would expect it to hit the db with a syntax like `ExperimentBuilder().fetch_config(cmdargs)`. And `ExperimentBuilder().fetch_config(cmdargs, use_db=False)` is pretty clear.